common.rkt

```diff
@@ -190,16 +237,34 @@
     ((pair? v) 1)
     (else 1)))
 
+;; checks to see if x contains any fresh statments
 (define (contains-fresh? x)
   (if (pair? x)
-      (or (contains-fresh? (car x)) (contains-fresh? (cdr x)))
-      (var? x)))
-(define (pretty-diseq =/=s) (map (lambda (=/=) (let ((x (car =/=)) (y (cdr =/=))) (if (term<? x y) (list x y) (list y x)))) =/=s))
+      (or (contains-fresh? (car x)) (contains-fresh? (cdr x))) ;; if a pair, check both sides
+      (var? x))) ;; if no pair then: if it's variable we must have a fresh otherwise we don't
+
+;; sorts by term comparison
+(define (pretty-diseq =/=s)
+  (map (lambda (=/=) (let ((x (car =/=)) (y (cdr =/=))) (if (term<? x y) (list x y) (list y x)))) =/=s))
+
+;; turns typed terms into pretty strings
 (define (pretty-types constraint) (list (type-check->sym (cdr constraint)) (car constraint)))
 
+;; returns string type for symbol, string, and number
 (define (type-check->sym pred)
   (cond
     ((eq? pred symbol?) 'sym)
     ((eq? pred string?) 'str)
     ((eq? pred number?) 'num)
+    (error "Invalid type")))
+
+;; turns typed terms into pretty strings
+(define (pretty-distypes constraint) (list (distype-check->sym (cdr constraint)) (car constraint)))
+
+;; returns string type for symbol, string, and number
+(define (distype-check->sym pred)
+  (cond
+    ((eq? pred symbol?) '!sym)
+    ((eq? pred string?) '!str)
+    ((eq? pred number?) '!num)
```

Comment:
Instead of `!sym`, use `not-sym` due to Scheme/Racket conventions

Reply:
https://github.com/DakshChan/first-order-miniKanren/commit/d5efb1eb9e2793c532e937f9bf75be51ffbea36f